includes/class-wc-countries.php

```diff
@@ -136,7 +138,7 @@ public function load_country_states() {
 		);
 
 		// Load only the state files the shop owner wants/needs.
-		$allowed = array_merge( $this->get_allowed_countries(), $this->get_shipping_countries() );
+		$allowed = array_merge( $this->get_allowed_countries( $filter_by_selling_location ), $this->get_shipping_countries() );
```

Comment:
Maybe 
```
$allowed = $filter_by_selling_location ? array_merge( $this->get_allowed_countries(), $this->get_shipping_countries() ) : $this->get_countries();
```
would work? Then `get_allowed_countries` wouldn't need to be modified as @claudiulodro suggested.